src/app/activityindicator/activityindicator.component.ts

```diff
@@ -0,0 +1,65 @@
+import { Component, OnInit } from '@angular/core';
+import { DashboardService } from '../services/dashboard.service';
+import { Router } from '@angular/router';
+
+@Component({
+    selector: 'app-activityindicator',
+    templateUrl: './activityindicator.component.html',
+    styleUrls: ['./activityindicator.component.scss']
+})
+export class ActivityindicatorComponent implements OnInit {
+
+    activityProgress: number;
+    curActivity: number;
+    activity: number;
+    stage: number;
+
+    indicatorClass = ['indblack', 'indblack', 'indblack'];
+
+    constructor(private _dashboardService: DashboardService, private _router: Router) { }
+
+    ngOnInit() {
+        this._dashboardService.getProgressStatus().subscribe(response => {
+            this.activity = response.activity;
+            this.stage = response.stage;
+            this.findRoute();
+        }, err => {
+            this._router.navigate(['/login']);
+        });
+    }
+
+    updateInd(x) {
+        if (x < this.activity) {
+            this.activityProgress = 3;
+        } else {
+            this.activityProgress = this.stage;
+        }
+
+        for (let i = 0; i < this.activityProgress; i++) {
+            this.indicatorClass[i] = 'indgreen';
+        }
+    }
+
+    findRoute() {
+        const module = this._router.url.split('/')[1];
+        switch (module) {
+            case ('introduction') : {
```

Comment:
again, the functions are essentially identical for each of the switch statements.
i'd use a dictionary to help reduce the amount of code.
example:
```
this.moduleDict = {
  'introduction': 1,
  'malaria101': 2,
  'meds': 3
};
// notice how this is very similar to an array
```
then, this entire function can be simplified to:
```
findRoute() {
  const module = this._router.url.split('/')[1];
  this.curActivity = this.moduleDict[module];
  this.updateInd(this.curActivity);
}
```